custom_components/zwave_mqtt/entity.py

```diff
@@ -147,17 +147,24 @@ def __init__(self, values):
     def value_changed(self, value):
         """Call when the value is changed."""
         if value.value_id_key in (v.value_id_key for v in self.values if v):
+            self.values_updated()
             self.async_write_ha_state()
 
     @callback
     def value_added(self):
         """Handle a new value for this entity."""
+        self.values_updated()
 
     @callback
     def instance_updated(self, new_status):
         """Call when the instance status changes."""
         self.async_write_ha_state()
 
+    @callback
+    def values_updated(self):
```

Comment:
Is this needed because we want to take the same action when either `value_added` or `value_changed` is called?

Reply:
Yeah, but I wanted to make it more generic (and fail safe) to prevent common mistakes such as forgetting to call the original super() method with common logic. This leaves the common value parsing in the base entity class.

Reply:
And yes, in this case we want to take the same action for value_added and value_changed

Reply:
Can we document more explicitly when a platform should use the `values_updated` callback and when it should use the other two `value_` callbacks? It's not clear to me when `values_updated` should be used and when `value_changed` should be used eg. Why do we name it `values` instead of `value`?

Reply:
Option 1:
Just leave it with value_changed / value_added and let each platform override this method if they see fit.

Option 2:
Make the value_added / value_changed methods private. 
Advertise "values_updated" or "value_updated" as the one to use within a platform if you like to be informed about the underlying data collection being updated.

Reply:
I think we should make it as strict as possible. So if option 2 gives enough flexibility, let's go for that.

Reply:
Although, since we're inheriting from this entity, subclasses will be able to override private methods too. So we should document if we don't want that.

Reply:
My personal preference goes to 2 with clear comments in the private methods. This will make the code easier to maintain/read in the longer end I think. You already see some duplicating going on in the platforms.

Reply:
Sounds good!

Reply:
Preference for a name ?
"value_updated"
"values_updated"
"entity_updated"
"entity_values_updated"
"on_value_update"

Reply:
I like `on_value_update`. That is more distinct from the other names we already have while keeping `value` in singular mode.

Reply:
OK, I've adjusted this. The only platform affected (atm) was the light platform so I kept the change here instead of separate PR.